examples/summarizePatches.py

```diff
@@ -0,0 +1,179 @@
+    if saveSummaries is not None:
+        expandedFileName = "expanded_" + saveSummaries
+        condensedFileName = "condensed_" + saveSummaries
+        expandedSummary = open(expandedFileName, 'w')
+        condensedSummary = open(condensedFileName, 'w')
+
+    # Default to all visits if none are specified
+    if visits is None:
+        visits = butler.queryMetadata('calexp_md', 'visit')
+
+    # Make BBoxes of each ccd in each visit
+    skymapBBoxes = {}
+    for visit in visits:
+        for ccd in camera:
+            ccdId = int(ccd.getSerial())
+            bbox = ccd.getBBox()
+            dataId = {'visit': visit, ccdKey: ccdId}
+            try:
+                md = butler.get('calexp_md', dataId)
+                wcs = afwImage.makeWcs(md)
+                ccdBBox = wcsBBox(bbox, wcs)
+                skymapBBoxes[(visit, ccdId)] = ccdBBox
+            except:
+                pass
+
+    # Run through patches and find overlaps with ccds
+    for tract in skymap:
+        for patch in tract:
+            visitsToCcds = {}
+            filtersToVisits = {}
+            numOverlaps = 0
+            patchId = patch.getIndex()
+            patchBBox = wcsBBox(patch.getInnerBBox(), tract.getWcs())
+
+            # Find which ccds the patch overlaps (if any)
+            for skymapId in skymapBBoxes:
+                if patchBBox.overlaps(skymapBBoxes[skymapId]):
+                    visitId, ccdId = skymapId
+                    if visitId not in visitsToCcds:
+                        visitsToCcds[visitId] = [ccdId]
+                        visitFilter = butler.queryMetadata('calexp_md', 'filter', visit=visitId)[0]
+                        if visitFilter not in filtersToVisits:
+                            filtersToVisits[visitFilter] = []
+                        filtersToVisits[visitFilter] += [visitId]
+                        numOverlaps += 1
+                    else:
+                        visitsToCcds[visitId] += [ccdId]
+                        numOverlaps += 1
+
+            # If there was an overlap, print and save information
+            if numOverlaps != 0:
+                print("patch=%r:    Found %r overlapping ccds." % (patchId, numOverlaps))
+
+                if saveOverview:
+                    overview.write("patch=" + str(patchId) + "\n")
+                    overview.write("    " + str(filtersToVisits) + "\n")
+
+                if saveSummaries is not None:
+                    for i_f, patchFilter in enumerate(filtersToVisits):
+
+                        # Write expanded file - one line for each filter in each patch
+                        expandedSummary.write("filter=" + patchFilter + " ")
+                        expandedSummary.write("tract=" + str(tract.getId()) + " ")
+                        expandedSummary.write("patch=")
+                        expandedSummary.write(str(patchId[0]) + "," + str(patchId[1]) + " ")
+                        for i_v, visit in enumerate(filtersToVisits[patchFilter]):
```

Comment:
You're right - that cleans it up a lot.  It's updated in a new version I just pushed.